src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -67,44 +67,43 @@ private function findAndSortTaggedServices(string|TaggedIteratorArgument $tagNam
             $reflector = null !== $class ? $container->getReflectionClass($class) : null;
             $checkTaggedItem = !$definition->hasTag($definition->isAutoconfigured() ? 'container.ignore_attributes' : $tagName);
 
-            foreach ($attributes as $attribute) {
```

Comment:
This loop is about looping over the various tags added explicitly in the service definition. Anything that skips this loop is broken by design. Explicit tag attributes must win over the AsTaggedItem attribute providing autoconfigured fallback values (see the priority of values in `PriorityTaggedServiceUtil::getDefault()`)